apps/omg_performance/test/omg_performance/integration/bazantine_event_test.exs

```diff
@@ -0,0 +1,145 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Performance.BazantineEventsTest do
+  use ExUnitFixtures
+  use ExUnit.Case, async: false
+  use OMG.ChildChain.Integration.Fixtures
+  use OMG.Watcher.Fixtures
+  alias OMG.Eth
+  require OMG.Utxo
+
+  alias OMG.Eth
+  alias OMG.Eth.RootChain
+  alias OMG.Performance
+  alias OMG.State.Transaction
+  alias OMG.Utxo
+  alias OMG.Watcher.HttpRPC.Client
+  alias OMG.Watcher.TestHelper
+
+  @moduletag :integration
+  @moduletag timeout: 70_000
+
+  @tag fixtures: [:watcher, :child_chain, :contract]
+  test "time response for asking for exit data", %{contract: %{contract_addr: contract}} do
+    dos_users = 10
+    ntx_to_send = 10
+    spenders = generate_users(2)
+
+    IO.puts("""
+    dos users: #{dos_users}
+    spenders: #{length(spenders)}
+    ntx_toxend: #{ntx_to_send}
+    exits per dos user: #{length(spenders) * ntx_to_send}
+    total exits: #{length(spenders) * ntx_to_send * dos_users}
+    """)
+
+    # TODO find another way to wait for data generations, maybe smoke_test_statistics
+    Performance.start_extended_perftest(ntx_to_send, spenders, contract)
+    Process.sleep(4000)
+
+    statistics =
+      Enum.map(1..dos_users, fn _ ->
+        Task.async(fn ->
+          stream_exit_data_blocks = generate_exit_info() |> Enum.map(&get_lazy_exit_data/1) |> Enum.to_list()
+          :timer.tc(fn -> stream_exit_data_blocks |> Enum.map(&Enum.to_list(&1)) end)
+        end)
+      end)
+      |> Enum.map(fn task ->
+        {time, exit_data_blocks} = Task.await(task, :infinity)
+
+        {time / 1_000,
+         exit_data_blocks
+         |> Enum.map(fn exits_utxo ->
+           Enum.reduce(exits_utxo, %{correct: 0, error: 0}, fn
+             %{"proof" => _}, stats -> Map.update(stats, :correct, 0, &(&1 + 1))
+             _, stats -> Map.update(stats, :error, 0, &(&1 + 1))
+           end)
+         end)
+         |> Enum.reduce(%{error: 0, correct: 0}, fn %{error: error, correct: correct}, statistics ->
+           Map.update(statistics, :error, 0, &(&1 + error))
+           |> Map.update(:correct, 0, &(&1 + correct))
+         end)}
+      end)
```

Comment:
This is _really_ difficult to grok and could probably be broken up better to make it more clear what's happening in this assignment. Could we break this up/refactor this? 😄